app/products/calls/types/calls.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type UserModel from '@typings/database/models/servers/user';
+
+export type CallsState = {
+    serverUrl: string;
+    myUserId: string;
+    calls: Dictionary<Call>;
+    enabled: Dictionary<boolean>;
+}
+
+export const DefaultCallsState = {
+    serverUrl: '',
+    myUserId: '',
+    calls: {} as Dictionary<Call>,
+    enabled: {} as Dictionary<boolean>,
+} as CallsState;
+
+export type Call = {
+	participants: Dictionary<CallParticipant>;
+    channelId: string;
+    startTime: number;
+    screenOn: string;
+    threadId: string;
+}
+
+export const DefaultCall = {
+    participants: {} as Dictionary<CallParticipant>,
+    channelId: '',
+    startTime: 0,
+    screenOn: '',
+    threadId: '',
+};
+
+export type CurrentCall = {
+    serverUrl: string;
+    myUserId: string;
+    participants: Dictionary<CallParticipant>;
+    channelId: string;
+    startTime: number;
+    screenOn: string;
+    threadId: string;
+    screenShareURL: string;
+    speakerphoneOn: boolean;
+}
+
+export type CallParticipant = {
+    id: string;
+    muted: boolean;
+    raisedHand: number;
+    userModel?: UserModel;
```

Comment:
It's optional because we don't always have it. We only have the UserModels once they've been injected in the enhanced function here: https://github.com/mattermost/mattermost-mobile/blob/5bb240dec880ee0b1165e4c10584df12d94bdaff/app/products/calls/screens/call_screen/index.ts#L26

If you remember, you suggested to do it this way. But I'd love to store them, so we don't have to waste the resources injecting them every time the currentCall changes. What do you think?

But this question made me realize we might have an edge case, where a user joins the call and we don't have their UserModel, so fixing that in: e0ba639bca81f08facf9bedf9ce6459b49b3c4dc

Reply:
ok got it, is fine to keep as optional then as you fixed the loading of missing profiles